### DISCUSSION TITLE
Problems with hmUI.showToast display in the simulator

### DISCUSSION THREAD

**Discussion by SashaCX75** [zepp-health/zeppos-forum#183] [Developer Feedback]
I'm trying to show a popup message in the simulator using the hmUI.showToast widget. If a watch model with ZeppOS 2.0 is selected in the simulator, everything works fine. But if the watch model with ZeppOS 1.0 is selected in the simulator, then a blank popup message is displayed, no text.

**Comment by MyoungXUE** [comment]
Hello, please provide more clues to help us troubleshoot the problem, thank you for your cooperation!

1. the complete call code
2. the simulator model used by Zepp OS 1.0 and Zepp OS 2.0

**Reply by zdunex25** [reply]
This is how **showToast** looks like on all available devices in simulator that runs ZeppOS 1.0:
`hmUI.showToast({ text: 'Tap Zones Enabled!' });`
![zepp_screenshot_1685111496524](https://github.com/zepp-health/zeppos-forum/assets/1674725/70ac242e-bbe9-473c-91e0-9ca988b463dd)

**Comment by SashaCX75** [comment]
I used the simplest code to check
`hmUI.showToast({text: "text"});`

This is how it looks on the GTR 3 simulator (and on the simulator of other models with ZeppOS 1)
![zepp_screenshot_1685285799778](https://github.com/zepp-health/zeppos-forum/assets/12464527/a5647661-55a1-44c7-9e6b-26288cdab450)

This is how it looks on the GTR 4 simulator (and on the simulator of other models with ZeppOS 2)
![zepp_screenshot_1685285752023](https://github.com/zepp-health/zeppos-forum/assets/12464527/cf088851-1f5d-42ff-94fc-a8769d15d892)

**Comment by MyoungXUE** [comment]
Hi, I verified that there is a problem with the showToast API on the GTR3 simulator, we will fix it as soon as possible.

**Comment by MyoungXUE** [comment]
@SashaCX75 
Hello, we fixed this issue in GTR3 v1.0.5 simulator, thanks again for feedback.
